### PR TITLE
Cached compares + DataFrame refactor

### PR DESCRIPTION
**Pull request by benartuso**
- When on CREATE calls to `compare`, we cache off a compare directory at `<repo>/.oxen/cache/compares/<compare_id>`, containing 4 dfs: `match.parquet`, `diff.parquet`, `left_only.parquet`, `right_only.parquet`. We also save off `LEFT` and `RIGHT` files indicating what commits were used to create the compare so we can tell when recalculation is needed if branch heads change.

- On GET for a given compare_id, we return the requested parent compare object (new) if the commit ids are consistent with what's in the cache. If the commit ids have changed since the last cache, we first recompute the compare object. 

- The new compare parent object contains urls at which the derived dfs (match.parquet, diff.parquet) can be queried with standard DFOpts and return a standard DF payload. 

- The standard df payload is refactored to contain separate nodes for `df` (which contains whole-df metadata (schema + size))  and `slice` (which has the request information given pagination. 

- This also fully deprecates the old **JsonDataFrameSlice** api in favor of the new `JsonDataFrameView`

### CODE REVIEW

**Comment by benartuso** [comment]
@lewazo

File: src/lib/src/view/compare.rs [hunk 1]
```diff
@@ -51,14 +55,115 @@ pub struct CompareEntriesResponse {
...
+        name: &str,
+        compare_id: Option<&str>,
+        left_commit_id: &str,
+        right_commit_id: &str,
+        df: DataFrame,
+        schema: Schema,
+    ) -> CompareDerivedDF {
+        let resource = compare_id.map(|compare_id| CompareVirtualResource {
+            url: format!(
+                "/compare/df/{}/{}/{}..{}",
```
Comment (gschoeni) [inline src/lib/src/view/compare.rs:151]: @benartuso let's update all these to /data_frame as the old /df is depreciated. Sorry for the confusion.

File: src/server/src/routes.rs [hunk 2]
```diff
@@ -134,10 +134,20 @@ pub fn config(cfg: &mut web::ServiceConfig) {
             "/{namespace}/{repo_name}/compare/file/{base_head:.*}",
             web::get().to(controllers::compare::file),
         )
-        // TODONOW: change this back to get if tesla allows.
+        .route(
+            "/{namespace}/{repo_name}/compare/df/{compare_id}/{path}/{base_head:.*}",
```
Comment (gschoeni) [inline src/server/src/routes.rs]: Update these to /data_frame

**Review by gschoeni — CHANGES_REQUESTED** [review]

File: src/lib/src/view/compare.rs (hunk 1 above)
Comment (benartuso) [reply src/lib/src/view/compare.rs:151]: 👍

File: src/server/src/routes.rs (hunk 2 above)
Comment (benartuso) [reply src/server/src/routes.rs]: 👍

**Pull request merged**